src/core/qzmq/src/qzmqsocket.cpp

```diff
@@ -421,15 +425,14 @@ class Socket::Private : public QObject
 		connect(sn_read, &QSocketNotifier::activated, this, &Private::sn_read_activated);
 		sn_read->setEnabled(true);
 
-		updateTimer = new QTimer(this);
-		connect(updateTimer, SIGNAL(timeout()), SLOT(update_timeout()));
+		updateTimer = new RTimer;
```

Comment:
have we considered using smart pointers for RTimer?

Reply:
ah, yes I think we can do that since RTimer doesn't require `deleteLater()`.